src/app/project/project-details-tab/project-details-tab.component.ts

```diff
@@ -169,13 +173,25 @@ export class ProjectDetailsTabComponent implements OnInit, AfterViewInit, OnDest
     // Disable mouse zoom on project details - iterferes with scrolling page.
     this.map.scrollWheelZoom.disable();
 
+    // Draw project-level shapefiles if they're available.
+    let shapefilesToDraw = this.shapefiles[0];
+    if (Array.isArray(this.project.shapefiles) && this.project.shapefiles.length > 0) {
+      shapefilesToDraw = this.project.shapefiles;
+      shapefilesToDraw = _.orderBy(shapefilesToDraw, ['order'], ['desc']);
+
+    }
 		// Convert documents to shapefiles
-		this.shapefileStyle = { color: this.project.shapeFileColour || Constants.style.DEFAULT_SHAPEFILE_COLOUR }; // Colour for the shapefiles
-		this.convertedShapefiles = this.shapefiles[0]?.map(sf => {
-			if (sf?._id && sf?.documentFileName?.length > 0) {
+		this.convertedShapefiles = shapefilesToDraw.map((sf: ProjectShapefile & Document) => {
+			if (sf?.documentFileName?.length > 0 && (sf?._id || sf?.document)) {
+        const fileId = sf?._id || sf?.document;
 				const escapedName = encode(sf.documentFileName).replace(/\(/g, '%28').replace(/\)/g, '%29').replace(/\\/g, '_').replace(/\//g, '_').replace(/\%2F/g, '_');
-				const shapeurl = this.pathAPI + '/document/' + sf._id + '/fetch/' + escapedName;
-				return new L.Shapefile(shapeurl, { isArrayBuffer: false, style: this.shapefileStyle });
+				const shapeurl = this.pathAPI + '/document/' + fileId + '/fetch/' + escapedName;
+				return new L.Shapefile(shapeurl, {
+          isArrayBuffer: false,
+          style: {
+            color: sf.colour || this.project.shapeFileColour || Constants.style.DEFAULT_SHAPEFILE_COLOUR
```

Comment:
Could be tidier to store this in a variable and leave it as a one liner here, instead of expanding the object.

Reply:
Sure I'll bind it